### PR TITLE
Simplified map names, XRT nickname, and latex names

### PR DESCRIPTION
**Pull request by wafels**
The map nickname property is intended to be the unique, commonly used name for an observatory-instrument-detector combination.  This PR fixes the XRT nickname by removing mention of the filter wheel settings.  Since the nickname is meant to uniquely identify observatory-instrument-detector combination it can be used to define the map name.

### CODE REVIEW

**Comment by Cadair** [comment]
I will restart travis once it's fixed.

File: sunpy/map/sources/hinode.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ def __init__(self, data, header, **kwargs):
         self.meta['telescop'] = "Hinode"
         self.meta['wavelnth'] = np.nan
         self.meta['waveunit'] = 'keV'
-        self._nickname = "{0} {1}-{2}".format(self.detector, fw1, fw2)
+        self._nickname = "XRT"
```
Comment (dpshelio) [inline sunpy/map/sources/hinode.py]: Shouldn't be the nikname the whole: XRT and detector? or you just want to show XRT without any other information?
Does this affect when plotting a map (as the title of the plot)?

**Comment by wafels** [comment]
XRT is the commonly used name for the soft X-ray imaging telescope on Hinode that uniquely defines it.  The spacecraft name Hinode is therefore superfluous for identification purposes.  The filter wheel combinations used in any particular observation should not be part of the nickname, but are better identified with the measurement.  
This PR will also change all the map plot titles that involve observational data.

**Comment by Cadair** [comment]
Unrelated but while on the subject of nickname it annoys me that we have things like "AIA 171 Angstroms" in the mpl title, rather than using the actual Angstrom symbol, or better yet converting into proper SI prefixes.

**Comment by wafels** [comment]
Turns out the AIA source also hard codes the nickname on line https://github.com/sunpy/sunpy/blob/master/sunpy/map/sources/sdo.py#46 . If you knock that line out then the nickname is defined as an empty string, even although the detector property is as expected - 'AIA'.  It seems that the default assignment in mapbase is not working correctly.

**Comment by Cadair** [comment]
I made the defaulting to detector a little more sane in wafels/sunpy#4 and also added a latex name for plots.

**Comment by wafels** [comment]
Astropy 1.0.4 renders the latex version of u.AA using u'$\\mathrm{\\mathring{A}}$', which raises a LaTeX error on my installation.  See also https://github.com/astropy/astropy/blob/master/astropy/units/si.py#L37 .  Anyone else have this problem?

File: sunpy/map/mapbase.py [hunk 2]
```diff
@@ -363,7 +363,10 @@ def name(self):
     @property
     def latex_name(self):
         """LaTeX formatted description of the Map."""
-        return self._base_name().format(measurement=self.measurement._repr_latex_())
+        if isinstance(self.measurement, u.Quantity):
+            return self._base_name().format(measurement=self.measurement._repr_latex_())
+        else:
+            return self._base_name().format(measurement=self.measurement)
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: just return self.name here.

File: sunpy/map/mapbase.py [hunk 3]
```diff
@@ -363,10 +363,7 @@ def name(self):
     @property
     def latex_name(self):
         """LaTeX formatted description of the Map."""
-        if isinstance(self.measurement, u.Quantity):
-            return self._base_name().format(measurement=self.measurement._repr_latex_())
-        else:
-            return self._base_name().format(measurement=self.measurement)
+        return self.name
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: This is not quite what I meant, this is:
```python
if isinstance(self.measurement, u.Quantity):
    return self._base_name().format(measurement=self.measurement._repr_latex_())
else:
    return self.name
```

**Comment by Cadair** [comment]
:+1:

**Comment by wafels** [comment]
Problem with re-writing the hashes....

================================ 14 failed, 986 passed, 24 skipped, 2 xpassed in 370.08 seconds ================================
All test files for figure hashes can be found in /tmp/tmpsmlkwfgz_figures
Traceback (most recent call last):
  File "/home/ireland/anaconda3/bin/py.test", line 6, in <module>
    sys.exit(py.test.main())
  File "/home/ireland/anaconda3/lib/python3.5/site-packages/_pytest/config.py", line 49, in main
    return config.hook.pytest_cmdline_main(config=config)
  File "/home/ireland/anaconda3/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/home/ireland/anaconda3/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/ireland/anaconda3/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "/home/ireland/anaconda3/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 596, in execute
    res = hook_impl.function(*args)
  File "/home/ireland/anaconda3/lib/python3.5/site-packages/_pytest/main.py", line 119, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "/home/ireland/anaconda3/lib/python3.5/site-packages/_pytest/main.py", line 115, in wrap_session
    config._ensure_unconfigure()
  File "/home/ireland/anaconda3/lib/python3.5/site-packages/_pytest/config.py", line 844, in _ensure_unconfigure
    self.hook.pytest_unconfigure(config=self)
  File "/home/ireland/anaconda3/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/home/ireland/anaconda3/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/ireland/anaconda3/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "/home/ireland/anaconda3/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 596, in execute
    res = hook_impl.function(*args)
  File "/home/ireland/projects/sunpy/sunpy/sunpy/conftest.py", line 70, in pytest_unconfigure
    json.dump(hash.hash_library, outfile, sort_keys=True, indent=4, separators=(',', ': '))
  File "/home/ireland/anaconda3/lib/python3.5/json/__init__.py", line 179, in dump
    fp.write(chunk)
TypeError: a bytes-like object is required, not 'str'

**Comment by Cadair** [comment]
huh @wafels that needs it's own issue by the looks of it.

**Comment by wafels** [comment]
Also, do the figure hash values themselves depend on which version of Python you're running?

**Comment by Cadair** [comment]
Yeah @wafels you should make the figure hashes under 2.7 for now.

**Comment by Cadair** [comment]
@wafels can you add a `CHANEGLOG` entry please, (add `[ci skip]` to the commit message).

:+1: 

@sunpy/sunpy-maintainers review please.

**Comment by dpshelio** [comment]
This will also need an update in: sunpy/sunpy-figure-tests - which I can do ones merged.

:+1:

**Pull request merged**